Sources/Sentry/SentryFileManager.m

```diff
@@ -669,19 +685,67 @@ - (void)createPathsWithOptions:(SentryOptions *_Nonnull)options
     self.envelopesPath = [self.sentryPath stringByAppendingPathComponent:EnvelopesPathComponent];
 }
 
-- (BOOL)createDirectoryIfNotExists:(NSString *)path error:(NSError **)error
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+NSString *
+sentryApplicationSupportPath(void)
 {
-    if (![[NSFileManager defaultManager] createDirectoryAtPath:path
-                                   withIntermediateDirectories:YES
-                                                    attributes:nil
-                                                         error:error]) {
-        *error = NSErrorFromSentryErrorWithUnderlyingError(kSentryErrorFileIO,
-            [NSString stringWithFormat:@"Failed to create the directory at path %@.", path],
-            *error);
-        return NO;
+    static NSString *sentryApplicationSupportPath;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        NSArray *paths = NSSearchPathForDirectoriesInDomains(
+            NSApplicationSupportDirectory, NSUserDomainMask, YES);
+        NSString *applicationSupportDirectory = [paths firstObject];
+        sentryApplicationSupportPath =
+            [applicationSupportDirectory stringByAppendingPathComponent:@"io.sentry"];
+
+        NSError *error;
+        if (!createDirectoryIfNotExists(sentryApplicationSupportPath, &error)) {
+            SENTRY_LOG_ERROR(
+                @"Failed to create directory %@: %@", sentryApplicationSupportPath, error);
+        }
+    });
+    return sentryApplicationSupportPath;
+}
+
+NSString *
+launchProfileMarkerPath(void)
+{
+    static NSString *sentryLaunchConfigPath;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sentryLaunchConfigPath =
+            [sentryApplicationSupportPath() stringByAppendingPathComponent:@"profileLaunch"];
+    });
+    return sentryLaunchConfigPath;
+}
+
+BOOL
+appLaunchProfileMarkerFileExists(void)
+{
+    return access(launchProfileMarkerPath().UTF8String, F_OK) == 0;
```

Comment:
`h`: This method calls into `sentryApplicationSupportPath`, which creates the `sentryApplicationSupportPath` if it doesn't exist. So, this code will create a new path for every first SDK launch. Instead, I think `sentryApplicationSupportPath` must only return the path without creating the directory. We can create the directory in `createPathsWithOptions` instead.

Reply:
Good call, need to separate out the side effect, and avoid doing that work on an app launch 👍🏻